drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1218 @@
+
+		retval = _tmag5273_int_gpio_config(dev, GPIO_INPUT);
+
+		if (retval < 0) {
+			LOG_DBG("Cannot configure GPIO |%d|", retval);
+			return -EINVAL;
+		}
+	}
+
+	uint8_t regdata = 0;
+
+	retval = i2c_reg_read_byte_dt(&drv_cfg->i2c,
+		TMAG5273_REG_DEVICE_CONFIG_2,
+		&regdata);
+
+	if (retval < 0) {
+		LOG_DBG("Could not read device config 2 register |%d|", retval);
+		return -EIO;
+	}
+
+	LOG_DBG("operation mode: %d", TMAG5273_GET_REG_BITS(regdata, TMAG5273_OPERATING_MODE));
+
+	/* check manufacturerer ID */
+	retval = i2c_reg_read_byte_dt(&drv_cfg->i2c,
+		TMAG5273_REG_MANUFACTURER_ID_LSB,
+		&regdata);
+
+	if (retval < 0) {
+		LOG_DBG("Could not read LSB of manufacturer id |%d|", retval);
+		return -EIO;
+	}
+
+	if (regdata != TMAG5273_MANUFACTURER_ID_LSB) {
+		LOG_DBG("uncexpected manufacturer id");
+		return -EINVAL;
+	}
+
+	retval = i2c_reg_read_byte_dt(&drv_cfg->i2c,
+		TMAG5273_REG_MANUFACTURER_ID_MSB,
+		&regdata);
+
+	if (retval < 0) {
+		LOG_DBG("Could not read MSB of manufacturer id |%d|", retval);
+		return -EIO;
+	}
+
+	if (regdata != TMAG5273_MANUFACTURER_ID_MSB) {
+		LOG_DBG("uncexpected manufacturer id");
+		return -EINVAL;
+	}
+
+	/* check device id*/
+	retval = i2c_reg_read_byte_dt(&drv_cfg->i2c,
+		TMAG5273_REG_DEVICE_ID,
+		&regdata);
+
+	if (retval < 0) {
+		LOG_DBG("Could not read device id |%d|", retval);
+		return -EIO;
+	}
+
+	tmag5273_read_device_status(dev);
+
+	retval = tmag5273_reset_device_status(dev);
+
+	if (retval < 0) {
+		LOG_DBG("Could not reset DEVICE_STATUS register |%d|", retval);
+		return -EIO;
+	}
+
+	drv_data->version = TMAG5273_GET_REG_BITS(regdata, TMAG5273_VER);
+
+	/* magnetic measurement range based on version, apply correct one */
+	if (drv_data->version == TMAG5273_VER_TMAG5273X1) {
+		drv_data->xyz_range = (drv_cfg->meas_range != TMAG5273_AXIS_RANGE_LOW)
+			? TMAG5273_MEAS_RANGE_HIGH_MT_VER1 : TMAG5273_MEAS_RANGE_LOW_MT_VER1;
+	} else if (drv_data->version == TMAG5273_VER_TMAG5273X2) {
+		drv_data->xyz_range = (drv_cfg->meas_range != TMAG5273_AXIS_RANGE_LOW)
+			? TMAG5273_MEAS_RANGE_HIGH_MT_VER2 : TMAG5273_MEAS_RANGE_LOW_MT_VER2;
+	} else {
+		LOG_DBG("unexpected version");
+		return -EINVAL;
+	}
+
+	/* deactivate interrupts if necessary (will be overwritten if needed) */
+	regdata = (TMAG5273_INT_MODE_NONE << TMAG5273_INT_MODE_POS)
+		| (drv_cfg->mask_intb_value << TMAG5273_INT_MASK_INTB_POS);
+
+	retval = i2c_reg_write_byte_dt(&drv_cfg->i2c,
+			TMAG5273_REG_INT_CONFIG_1,
+			regdata);
+
+	if (retval < 0) {
+		LOG_DBG("Error deactivating interrupts");
+		return -EIO;
+	}
+
+	/* set settings */
+	retval = tmag5273_init_sensor_settings(drv_cfg);
+
+	if (retval < 0) {
+		LOG_DBG("error setting sensor configuration |%d|", retval);
+		return retval;
+	}
+
+	retval = tmag5273_init_device_config(drv_cfg);
+
+	if (retval < 0) {
+		LOG_DBG("error setting device configuration |%d|", retval);
+		return retval;
+	}
+
+	return 0;
+}
+
+/* ----------------------------------------------------------- */
```

Comment:
Please remove these separators throughout the driver. It's already extremely long and these lines just make it longer